controllers/secret_controller.go

```diff
@@ -65,13 +65,8 @@ func NewSecretReconciler(mgr manager.Manager, clusterConfig cluster.Config,
 
 // SetupWithManager sets up a new Secret controller
 func (r *SecretReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
-	// Check that the private key exists, if it doesn't, log a warning
-	_, err := secrets.GetPrivateKey(ctx, kubeTypes.NamespacedName{Namespace: r.watchNamespace,
-		Name: secrets.PrivateKeySecret}, mgr.GetClient())
-	if err != nil {
-		r.log.Error(err, "Unable to retrieve private key, please ensure it is created")
-	}
-
+	// cache is not available during setup, hence won't check for private key
+	// the Reconcile will handle missing private key scenarios properly
```

Comment:
This comment doesn't really make sense without the context of the removed code.
Moving `the Reconcile will handle missing private key scenarios properly` into the commit message would be good.